interfaces/Portalicious/src/app/pages/project/project-registrations/project-registration-personal-information/project-registration-personal-information.page.ts

```diff
@@ -18,13 +38,70 @@ import { PageLayoutComponent } from '~/components/page-layout/page-layout.compon
     CommonModule,
     FormsModule,
     SelectButtonModule,
+    InputTextModule,
+    FormDefaultComponent,
+    FormFieldWrapperComponent,
+    DataListComponent,
   ],
   templateUrl: './project-registration-personal-information.page.html',
   styles: ``,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ProjectRegistrationPersonalInformationPageComponent {
-  // this is injected by the router
+  readonly registrationApiService = inject(RegistrationApiService);
+  readonly projectApiService = inject(ProjectApiService);
+  readonly fspApiService = inject(financialServiceProviderApiService);
+
   projectId = input.required<number>();
   registrationId = input.required<number>();
+
+  projectAttributes = injectQuery(
+    this.projectApiService.getProjectAttributes({
+      projectId: this.projectId,
+      includeCustomAttributes: true,
+      includeFspQuestions: true,
+      includeProgramQuestions: true,
+      includeTemplateDefaultAttributes: true,
+    }),
+  );
+
+  registration = injectQuery(
+    this.registrationApiService.getRegistrationById(
+      this.projectId,
+      this.registrationId,
+    ),
+  );
+
+  selectedFspQuestionList = injectQuery(() => ({
+    ...this.fspApiService.getFinancialServiceProviderQuestions(
+      this.registration.data()?.financialServiceProvider,
+    )(),
+    enabled:
+      this.registration.isSuccess() && this.projectAttributes.isSuccess(),
+  }));
+
+  attributeList = computed<DataListItem[]>(() => {
+    const list: DataListItem[] = [];
+    if (!this.projectAttributes.isSuccess() || !this.registration.isSuccess()) {
+      return list;
+    }
+
+    for (const attribute of this.projectAttributes.data()) {
+      if (
+        attribute.questionType === QuestionType.fspQuestion &&
+        !this.selectedFspQuestionList.data()?.includes(attribute.name)
+      ) {
+        continue;
+      }
+
+      // XXX: where do we get the value for custom attributes? eg. Westeros -> house
```

Comment:
It is indeed in `this.registration.data()[attribute.name]`, however, the key `attribute.name` might not exist yet if the content is not defined. This might change in Ruben's refactor.